core/modules/czar/MessageTable.cc

```diff
@@ -68,6 +70,19 @@ MessageTable::MessageTable(std::string const& tableName,
       _sqlConn(std::make_shared<sql::SqlConnection>(resultConfig)) {
 }
 
+// Create the table, do not lock
+void
+MessageTable::create() {
```

Comment:
From API point of view these methods should be separate, the client code knows which method to call at compile time, there is no run-time decision to whether lock or not. I don't care too much about code duplication, it's not that terrible, that function only does absolutely minimal job to execute the query, there is no heavy logic there.